src/optree.cpp

```diff
@@ -234,11 +251,17 @@ void BuildModule(py::module& mod) {  // NOLINT[runtime/references]
 #ifdef Py_TPFLAGS_IMMUTABLETYPE
     reinterpret_cast<PyTypeObject*>(PyTreeSpecTypeObject.ptr())->tp_flags |=
         Py_TPFLAGS_IMMUTABLETYPE;
+    reinterpret_cast<PyTypeObject*>(PyTreeKindTypeObject.ptr())->tp_flags |=
```

Comment:
duplicated?

Reply:
The type objects are different. They are `PyTreeSpec` and `PyTreeKind`. Here we set both two types immutable on the Python side.

Reply:
Oh yes, I see.